binder/src/main/java/io/grpc/binder/internal/BinderTransportSecurity.java

```diff
@@ -198,6 +204,18 @@ ListenableFuture<Status> checkAuthorization(MethodDescriptor<?, ?> method) {
     }
```

Comment:
Did you consider just adding a callback here that called `serviceAuthorization.remove(serviceName, authorization)` from onFailure()  ?

Reply:
Done.

Reply:
Thanks for switching your remove approach but I find the new catching() future confusing and unconventional. The future returned by catching() is ignored and you are doing the remove work as a side effect of what's supposed to be an exception transformation function ... Can you consider Futures.addCallback() with a remove call in onFailure() instead?

Reply:
Done. I also reverted back to ConcurrentHashMap to avoid dealing with #10670. I may need to think over that one a bit more, but it may be a non-issue since we are using `putIfAbsent`, so "read-modify-write" races may not actually be as possible as I originally thought.